tests/hpc/hpc_master.pm

```diff
@@ -0,0 +1,99 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: HPC master
+#    This test is aiming at comprehensive HPC testing. That means
+#    that most components of the HPC product should be installed and
+#    configured. At the same time it is expected that the HPC cluster
+#    set-up is fairly complex, so that NFS shares are mounted, required
+#    database(s) are ready to be used, some data is being populated to
+#    the databases, artificial users are added etc.
+# Maintainer: Sebastian Chlad <sebastian.chlad@suse.com>
+
+use base "hpcbase";
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils;
+
+sub mount_nfs {
+    ## TODO: move to the lib
+    systemctl 'start nfs';
+    systemctl 'start rpcbind';
+    record_info('show mounts aviable on the supportserver', script_output('showmount -e 10.0.2.1'));
+    assert_script_run("mkdir -p /shared/slurm");
+    assert_script_run("chown -Rcv slurm:slurm /shared/slurm");
+    assert_script_run("mount -t nfs -o nfsvers=3 10.0.2.1:/nfs/shared /shared/slurm");
+}
+
+sub run {
+    my $self  = shift;
+    my $nodes = get_required_var("CLUSTER_NODES");
+    $self->prepare_user_and_group();
+
+    # provision HPC cluster, so the proper rpms are installed,
+    # and all the set-up is done, including external services, like NFS etc.
+    record_info('Start installing all components');
+    zypper_call('in slurm slurm-munge ganglia-gmetad ganglia-gmond ganglia-gmetad-skip-bcheck');
+    zypper_call('in nfs-client rpcbind');
+    record_info('Installation done');
+
+    record_info('System set-up: prepare needed configuration');
+    mount_nfs();
+    $self->prepare_slurm_conf();
+    record_info('slurmctl conf', script_output('cat /etc/slurm/slurm.conf'));
+    barrier_wait("HPC_SETUPS_DONE");
+
+    record_info('System set-up: enable & start services');
+    $self->distribute_munge_key();
+    $self->distribute_slurm_conf();
+    $self->enable_and_start('munge');
+    $self->enable_and_start('slurmctld');
+    systemctl 'status slurmctld';
+    $self->enable_and_start('slurmd');
+    systemctl 'status slurmd';
+    $self->enable_and_start('gmetad');
+    systemctl 'status gmetad';
```

Comment:
You are mixing systemctl 'status gmetad'; and systemctl("status gmetad").
Could you please pick one style and keep it consistent? Should we define some coding style here?

Reply:
should be changed but easy to overlook :(